scripts/staging/hmm/HMM Training.py

```diff
@@ -130,6 +203,7 @@ def train_markov_model(data):
 initial_word = {}
 second_word = {}
 transitions = {}
-train_markov_model(data)
-sentence = 'cocoa bahia'
-generate(sentence,2)
+#Second parameter is to determine how many previous words algorithm takes when learning
+train_markov_model_generic(data, 5)
```

Comment:
Hi Afan,
Thank you for solving issues in data cleaning script. Now the training script is sucked in an infinite loop after making this call train_markov_model_generic(data, 5). It never stops executing. 
Please fix this so that I can test your script. 
Thank you.

Reply:
@Shafaq-Siddiqi It is not an infinite loop, but the whole learning algorithm is very slow because it takes all possible combinations of words and calculates probabilities. That is the purpose of parameter no_of_words in method train_markov_model_generic.

Reply:
As already discussed, please restrict your maximum k (no_of_words = 3). If a user provides any number greater than 3 just print a message and call the script with  no_of_words = 3. It is not only slow but also consuming memory.  Please restrict your input/output parameters and include validation checks like if sentence length is 0 and no_of_words > sentence length etc.